addon/services/engine-router-service.js

```diff
@@ -0,0 +1,121 @@
+import Ember from 'ember';
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { computed } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+
+// @HACK: since `getEngineParent` is not exported
+function getEngineParent(engine) {
+  const symbolPrefix = `__ENGINE_PARENT${Ember.GUID_KEY}`;
+  const symbol = Object.keys(engine).find(k => k.startsWith(symbolPrefix));
+  if (!symbol) {
+    return null;
+  }
+  return engine[symbol];
+}
+
+export default Service.extend({
+  engine: computed(function () {
+    return getOwner(this);
+  }),
+
+  externalRoutes: reads('engine._externalRoutes'),
+
+  mountPoint: reads('engine.mountPoint'),
+
+  rootURL: reads('externalRouter.rootURL'),
+
+  currentURL: reads('externalRouter.currentURL'),
+
+  currentRouteName: computed('externalRouter.currentRouteName', function () {
+    if (this.externalRouter.currentRouteName === this.mountPoint) {
+      return 'application';
+    }
+    return this.externalRouter.currentRouteName.slice(this.mountPoint.length + 1);
+  }),
+
+  externalRouter: computed(function () {
+    return this.rootApplication.lookup('service:router');
+  }),
+
+  rootApplication: computed('engine', function () {
+    let parent = getEngineParent(this.engine);
+    while (getEngineParent(parent)) {
+      parent = getEngineParent(parent);
+    }
+    return parent;
+  }),
+
+  getInternalRouteName(internalRouteName) {
+    // https://github.com/ember-engines/ember-engines/blob/ec4d1ae7a413a7e5d9e57a4e3b2e0f0d19a0afcd/addon/components/link-to-component.js#L52-L57
```

Comment:
The referenced function should be made into a util fucntion like:

```
export function namespaceEngineRouteName(mountPoint, routeName) {
  if (routeName === 'application') {
    return mountPoint;
  } else {
    return `${this.mountPoint}.${routeName}`;
  }
}
```

Then used whereever `getInternalRouteName` is used in this PR (and used in `link-to` at the location this comment points to)